acceptance/lib/puppet/acceptance/solaris_util.rb

```diff
@@ -87,8 +87,15 @@ def set_publisher(agent, o={})
     module SMFUtils
       def clean(agent, o={})
         o = {:service => 'tstapp'}.merge(o)
-        on agent, "svcadm disable %s ||:" % o[:service]
-        on agent, "svccfg delete %s ||:" % o[:service]
+        on(agent, "svcs -l %s" % o[:service], acceptable_exit_codes: [0, 1]) do |result|
+          next if result.stdout =~ /doesn't match/
+          lines = result.stdout.chomp.lines
+          instances = lines.select { |line| line =~ /^fmri/ }.map { |line| line.split(' ')[1].chomp }
+          instances.each do |instance|
+            on agent, "svcadm disable -s %s ||:" % instance
+            on agent, "svccfg delete %s ||:" % instance
+          end
+        end
```

Comment:
so this is walking all the FMRIs?

Reply:
Yes, of the given service b/c I have an acceptance test that adds an extra instance to check that Puppet fails if the service resource corresponds to multiple instances.